examples/platforms/kw41z/radio.c

```diff
@@ -745,39 +778,36 @@ void Radio_1_IRQHandler(void)
         switch (state)
         {
         case XCVR_RX_c:
-            temp = (ZLL->LQI_AND_RSSI & ZLL_LQI_AND_RSSI_LQI_VALUE_MASK) >> ZLL_LQI_AND_RSSI_LQI_VALUE_SHIFT;
-            sRxFrame.mLength = (ZLL->IRQSTS & ZLL_IRQSTS_RX_FRAME_LENGTH_MASK) >> ZLL_IRQSTS_RX_FRAME_LENGTH_SHIFT;
-            sRxFrame.mLqi = rf_lqi_adjust(temp);
-            sRxFrame.mPower = rf_lqi_to_rssi(sRxFrame.mLqi);
-#if DOUBLE_BUFFERING
-            memcpy(sRxData, (void *)ZLL->PKT_BUFFER_RX, sRxFrame.mLength);
-#endif
-            sRxDone = true;
+            sRxDone = rf_process_rx_frame();
             break;
 
-        case XCVR_TX_c:
         case XCVR_TR_c:
+            if (!(irqStatus & ZLL_IRQSTS_RXIRQ_MASK) ||
+                (rf_process_rx_frame() == false) ||
+                (sRxFrame.mLength != IEEE802154_ACK_LENGTH) ||
+                ((sRxFrame.mPsdu[IEEE802154_FRM_CTL_LO_OFFSET] & IEEE802154_FRM_TYPE_MASK) != IEEE802154_FRM_TYPE_ACK) ||
+                (sRxFrame.mPsdu[IEEE802154_DSN_OFFSET] != sTxFrame.mPsdu[IEEE802154_DSN_OFFSET]))
+            {
+                sTxStatus = OT_ERROR_NO_ACK;
+            }
+
```

Comment:
If a channel access failure is encountered by the radio, then the TR sequence (TX followed by an RX) will end, and the RXIRQ flag cannot be asserted. The NO_ACK and CHANNEL_ACCESS_FAILURE status codes are exclusive.
Do you want me to rewrite this piece of code to make it more readable?